### ISSUE TITLE
limit(6^x, x, 1) and similar cases: stack overflow crash

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#4085]
Imported from SourceForge on 2024-07-09 11:18:45
Created by **[tomasriker](https://sourceforge.net/u/tomasriker/)** on 2023-12-15 16:19:15
Original: https://sourceforge.net/p/maxima/bugs/4222

---

Integrating, with limits, expressions of the type `1/n^x` with `n` integer, often makes Maxima crash, for example:

~~~
 -->	integrate(1/10^x, x, 0, 1);
Binding stack guard page temporarily disabled: proceed with caution
~~~

After some more investigation, the cause is a simple limit like this, which makes Maxima crash:

~~~
 -->	limit(6^x, x, 1);
 Binding stack guard page temporarily disabled: proceed with caution
~~~

What it comes down to is this:

~~~
 -->	limit(2^x * 3^x, x, 1);
 Binding stack guard page temporarily disabled: proceed with caution
~~~

The commit that introduced this behavior is [20b3068fb7e8f3528e9b6aa7abf47185248cb8e3].

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-09 11:18:47
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2023-12-18 03:10:28
Original: https://sourceforge.net/p/maxima/bugs/4222/#e36b

---

I have a fix for these bugs. The fix changes the predicate  `expfactorp` to return false on any expression that involves an extended real (minf, inf, ....). I'll check it in later this week.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-09 11:18:54
Created by **[tomasriker](https://sourceforge.net/u/tomasriker/)** on 2023-12-19 21:21:38
Original: https://sourceforge.net/p/maxima/bugs/4222/#804f

---

Fixed by [82ace5a38f201e3006dc8917caa8670c863ad6d6]
